server/src/main/java/org/hyperskill/community/flashcards/common/ActionsParser.java

```diff
@@ -0,0 +1,32 @@
+package org.hyperskill.community.flashcards.common;
+
+import org.hyperskill.community.flashcards.common.response.ActionType;
+import org.hyperskill.community.flashcards.common.response.PermittedAction;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ActionsParser {
+    private static String currentUri = "";
+
+    private ActionsParser() {
+    }
+
+    public static void setUri(String uri) {
+        currentUri = uri;
+    }
+
+    public static Set<PermittedAction> fromPermissions(String permissions) {
+        Set<PermittedAction> actions = new HashSet<>(3);
+        if (permissions.contains("r")) {
+            actions.add(new PermittedAction(ActionType.READ, currentUri));
+        }
+        if (permissions.contains("w")) {
+            actions.add(new PermittedAction(ActionType.WRITE, currentUri));
+        }
+        if (permissions.contains("d")) {
+            actions.add(new PermittedAction(ActionType.DELETE, currentUri));
+        }
+        return actions;
```

Comment:
why don't you just hand the uri as parameter to fromPermissions?
Now you MUST call setUri before the fromPermissions call - otherwise it will use an old variable. It is error prone.
My suggestion: drop this static state variable (actually I don't typically use a class state...) and have it a local variable in the Utils call.